tests/app/celery/test_broadcast_message_tasks.py

```diff
@@ -471,3 +581,137 @@ def test_trigger_link_tests_invokes_cbc_proxy_client(
+        broadcast_message,
+        transmitted_starts_at=datetime(2021, 1, 1, 0, 0),
+        transmitted_finishes_at=datetime(2021, 1, 1, 11, 59),
+    )
+    with pytest.raises(CBCProxyFatalException) as exc:
+        check_provider_message_should_send(current_event, 'ee')
+    assert 'The expiry time of 2021-01-01 11:59:00 has already passed' in str(exc.value)
+
+
+@freeze_time('2021-01-01 12:00')
+def test_check_provider_message_should_send_raises_if_older_event_still_sending(sample_template):
+    broadcast_message = create_broadcast_message(sample_template)
+    # event approved at midnight
+    past_succesful_event = create_broadcast_event(
+        broadcast_message,
+        message_type='alert',
+        sent_at=datetime(2021, 1, 1, 0, 0),
+    )
+    # event updated at 5am (this is the event we're currently trying to send)
+    past_still_sending_event = create_broadcast_event(
+        broadcast_message,
+        message_type='update',
+        sent_at=datetime(2021, 1, 1, 5, 0),
+    )
+    # event updated at 7am
+    current_event = create_broadcast_event(
+        broadcast_message,
+        message_type='update',
+        sent_at=datetime(2021, 1, 1, 7, 0),
+    )
+
+    create_broadcast_provider_message(past_succesful_event, provider='ee', status=BroadcastProviderMessageStatus.ACK)
+    create_broadcast_provider_message(past_still_sending_event, provider='ee', status=BroadcastProviderMessageStatus.SENDING)  # noqa
+
+    # we havent sent the previous update yet - it's still in sending - so don't try and send this one.
+    with pytest.raises(CBCProxyFatalException) as exc:
+        check_provider_message_should_send(current_event, 'ee')
+
+    assert f'Previous event {past_still_sending_event.id} (type update) has not finished sending to provider ee' in str(exc.value)  # noqa
+
+
+@freeze_time('2021-01-01 12:00')
+def test_check_provider_message_should_send_raises_if_older_event_hasnt_started_sending_yet(sample_template):
+    broadcast_message = create_broadcast_message(sample_template)
+    # event approved at midnight
+    past_succesful_event = create_broadcast_event(
+        broadcast_message,
+        message_type='alert',
+        sent_at=datetime(2021, 1, 1, 0, 0),
+    )
+    # event updated at 5am (this is the event we're currently trying to send)
+    past_still_sending_event = create_broadcast_event(
+        broadcast_message,
+        message_type='update',
+        sent_at=datetime(2021, 1, 1, 5, 0),
+    )
+    # event updated at 7am
+    current_event = create_broadcast_event(
+        broadcast_message,
+        message_type='update',
+        sent_at=datetime(2021, 1, 1, 7, 0),
+    )
+
+    # no provider message for past_still_sending_event
+    create_broadcast_provider_message(past_succesful_event, provider='ee', status=BroadcastProviderMessageStatus.ACK)
+
+    # even though the task is going on until midnight tomorrow, we shouldn't send the update now, because the cancel
```

Comment:
`cancel message` comment seems irrelevant now? There are no cancels, just updates. Would suggest changing this comment to:

this task shouldn't send because the `past_still_sending_event` hasn't started sending yet for this provider